kebechet/config.py

```diff
@@ -46,36 +46,80 @@ def from_file(self, config_path: str):
                 content = config_file.read()
 
         try:
-            self._repositories = yaml.safe_load(content).pop('repositories') or []
+            self._repositories = yaml.safe_load(content).pop("repositories") or []
         except Exception as exc:
-            raise ConfigurationError("Failed to parse configuration file: {str(exc)}") from exc
+            raise ConfigurationError(
+                "Failed to parse configuration file: {str(exc)}"
+            ) from exc
+
+    def download_conf_from_url(self, url: str, service: str):
+        params = url.split("/")
+        slug = "/".join(params[3:])
+        user = params[3]
+        repo = params[4]
+        _LOGGER.info(f"{service} service detected")
+        if service == "github":
+            token = os.environ["GITHUB_TOKEN"]
+            download_uri = (
+                f"https://raw.githubusercontent.com/{slug}/master/.kebechet.yaml"
+            )
+            _LOGGER.info(f"Downloading config from: {download_uri}")
+            resp = requests.get(
+                download_uri, headers={"Authorization": f"token {token}"}
+            )
+        elif service == "gitlab":
+            token = os.environ["GITLAB_TOKEN"]
+            download_uri = (
+                f'https://gitlab.com/api/v4/projects/{slug.replace("/", "%2F")}' +
+                '/repository/files/.kebechet.yaml/raw?ref=master'
+            )
+            _LOGGER.info(f"Downloading config from: {download_uri}")
+            resp = requests.get(download_uri, headers={"Private-Token": token})
+        elif service == "pagure":
+            pass  # TODO
+        else:
+            _LOGGER.warning("Service not supported")
+
+        open("temp.yaml", "wb").write(resp.content)
```

Comment:
Please use the context manager. The process will still have opened this file.

It will be a good idea to utilize [NamedTemporaryFile](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile) for this - you can manage file's lifecycle in `run_url()` method. As this will run in a container, the $PWD of the process does not need to be writable.